pallets/idn-manager/src/lib.rs

```
@@ -403,6 +413,11 @@ pub mod pallet {
 		) -> DispatchResultWithPostInfo {
 			let subscriber = ensure_signed(origin)?;
 
+			ensure!(
+				SubCounter::<T>::get() < T::MaxSubscriptions::get(),
```

If max subscriptions is 1M, does that mean that we want to throw an error at 1M and not at 1M + 1?

---

this check happens before creating the subscription, so if the limit is 1M and we are almost full when calling `create_subscription()` this assertion will check that the count is 999,999 before letting the last one in to complete the 1M limit